### PR TITLE
Add fullname to owner on logging

### LINKED ISSUE #29662: Audit Log is unclear when using Azure AD login
### Apache Airflow version

2.5.1

### What happened

We're using an Azure OAUTH based login in our Airflow implementation, and everything works great. This is more of a visual problem than an actual bug.

In the Audit logs, the `owner` key is mapped to the username, which in most cases is airflow. But, in situations where we manually pause a DAG or enable it, it is mapped to our generated username, which doesn't really tell one who it is unless they were to look up that string in the users list. Example:

image

It would be nice if it were possible to include the user's first and last name alongside the username. I could probably give this one a go myself, if I could get a hint on where to look.

I've found the dag_audit_log.html template, but not sure where to change log.owner. 

### What you think should happen instead

It would be good to get a representation such as username (FirstName LastName).

### How to reproduce

N/A

### Operating System

Debian GNU/Linux 11

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Deployed with Helm chart v1.7.0, and Azure OAUTH for login.

### Anything else

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by arjunanan6**
closes: #29662 

This change adds the owner's full name in brackets next to the owner username.

### CODE REVIEW

**Comment by arjunanan6** [comment]
@potiuk Seems like 1 test out of 40 fails, but I cannot see which. Would you be able to help?

**Comment by malthe** [comment]
This looks great, but it would be nice to have a means of providing a formatting function in `$AIRFLOW_HOME/webserver_config.py`.

**Comment by uranusjr** [comment]
Instead of changing the meaning of `user`, it may be better to add a new field for the full name.

**Comment by arjunanan6** [comment]
@uranusjr There is a screenshot in my linked issue, the reason is to just make it clear in the Audit logs as to who has done what. At least in our case, it just references the azure ID which doesn't say anything.

Could you give me a hint on where to introduce the JOIN? I could take a look at it then.

**Comment by arjunanan6** [comment]
@uranusjr Actually, figured out the addition of a new field to the log table, I was just overthinking it. I'll put some more into it and return :) 

This is what it looks like:
image

File: airflow/www/decorators.py [hunk 1]
```diff
@@ -87,7 +87,8 @@ def wrapper(*args, **kwargs):
                 if g.user.is_anonymous:
                     user = "anonymous"
                 else:
-                    user = g.user.username
+                    full_name = g.user.get_full_name()
+                    user = g.user.username + f" ({full_name})"
```
Comment (uranusjr) [inline airflow/www/decorators.py]:
```suggestion
                    user = f"{g.user.username} ({g.user.get_full_name()})"
```

**Comment by arjunanan6** [comment]
Hi @uranusjr, I've commited the suggested change. Is it good to go?

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Doc building failure is unrelated.

**Comment by arjunanan6** [comment]
@uranusjr Should be good to merge now :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**